### PR TITLE
[NUI] Particle System implementation

### PR DESCRIPTION
**Pull request by jsr184**
### Description of Change ###
Implementation of the Particle System for NUI (binding to the native side + C# specific implementation of certain elements).

Classes included:
Tizen.NUI.ParticleSystem.ParticleEmitter
Tizen.NUI.ParticleSystem.ParticleSource<T>
Tizen.NUI.ParticleSystem.ParticleSourceInterface
Tizen.NUI.ParticleSystem.ParticleModifier<T>
Tizen.NUI.ParticleSystem.ParticleModifierInterface
Tizen.NUI.ParticleSystem.Particle

ParticleSource and ParticleModifier invoke callbacks on the update thread so the implementation of those callbacks should not be accessing NUI objects.

### CODE REVIEW

**Review by adeelkazmi — APPROVED** [review]

**Review by adeelkazmi — CHANGES_REQUESTED** [review]
There are 5 commits in this change including temporary ones. Please merge into 1.

**Review by adeelkazmi — APPROVED** [review]

File: src/Tizen.NUI/src/public/ParticleSystem/ParticleEmitter.cs [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+using System.ComponentModel;
+
+namespace Tizen.NUI.ParticleSystem
+{
+    using Tizen.NUI.BaseComponents;
+
+    /// <summary>
+    /// Enum defining blending options when rendering the particles.
+    /// </summary>
+    public enum ParticleBlendingMode
```
Comment (hinohie) [inline src/Tizen.NUI/src/public/ParticleSystem/ParticleEmitter.cs:30]:
```suggestion
    [EditorBrowsable(EditorBrowsableState.Never)]
    public enum ParticleBlendingMode
```

We should also mark public enums as 'hidden' if we are not prepare to ACR.

File: src/Tizen.NUI/src/public/ParticleSystem/ParticleEmitter.cs [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+    /// View. ParticleEmitter is responsible for spawning and updating particles.
+    ///
+    /// Emitter must contain:
+    /// ParticleSource - responsible for spawning new particles
+    /// ParticleModifier(s) - responsible for updating particles in the system
+    ///
+    /// ParticleSource and ParticleModifier callback interfaces should not be accessing
+    /// Event side (NUI) objects. Both callbacks are executed on Update thread.
+    /// </summary>
+    public class ParticleEmitter : BaseHandle
```
Comment (hinohie) [inline src/Tizen.NUI/src/public/ParticleSystem/ParticleEmitter.cs:63]:
```suggestion
    [EditorBrowsable(EditorBrowsableState.Never)]
    public class ParticleEmitter : BaseHandle
```

Please make this class as hidden, if you don't prepare ACR.

**Comment by hinohie** [comment]
It is just our NUI team's usual process but...

Usually  ACR process spend more than 3~4 month.
So, We make Implement PR with internal-hidden only APIs first.
And after that PR merged, make new PR only for ACR.

https://github.com/Samsung/TizenFX/pull/5452/files

Please check what I'm doimg now. and also please make some API as hidden, what I write comment to you

**Comment by dongsug-song** [comment]
@jsr184 
Hello, 
as Mr. hinohie explained above, please add `[EditorBrowsable(EditorBrowsableState.Never)]` on every API to make them hidden APIs. and to release them in public, you need to procede ACR. 
please check again. 
thank you.

**Comment by hinohie** [comment]
PR #5548 is more latest version. We will close this PR.

**Pull request closed without merging**